app/src/main/java/com/sdp/movemeet/map/MiniMapFragment.java

```diff
@@ -0,0 +1,70 @@
+package com.sdp.movemeet.map;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import androidx.core.app.ActivityCompat;
+import androidx.fragment.app.Fragment;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.android.gms.tasks.Task;
+import com.sdp.movemeet.R;
+import com.sdp.movemeet.UploadActivityActivity;
+import com.sdp.movemeet.utility.LocationFetcher;
+
+public class MiniMapFragment extends Fragment implements OnMapReadyCallback, GoogleMap.OnMapClickListener {
```

Comment:
What is actually the difference between MiniMapFragment and the other main map fragment? Does MiniMapFragment only show the nearby activities, but we can't create a new activity directly from it?

Reply:
The mini map is used in the UploadActivity to let the user choose the location on click. The principal difference is that it doesn't display the activities, and directly sets the address of the location clicked. Whereas the main fragment is displayed in the main activity, to display all activities and with the possibility for the user to go to the UploadActivity fragment by clicking on where he wants to set his activity. It's quite similar in the idea, but still have different behaviors, hence the necessity to use two different classes.